### PR TITLE
New template `key_value_pair_in_file` and audispd remote daemon rules

### PR DESCRIPTION
**Pull request by yuumasato**
#### Description:

- Add new template for `key = value` in file.
  - This template covers a different use case than `lineinfile` template.
  - `key_value_pair_in_file` can correct confl
- Add rules for `audispd` that use the template

#### Rationale:

- Template `lineinfile` doesn't cover the audispd rules very well.
  - The audispd config files already contain a line `active = no` and `lineinfile` template would add a new line `active = yes` instead of updating the `active` key with value `yes`.

#### Review Hints:

- `python3  tests/automatus.py rule --libvirt qemu:///session rhel79 --datastream build/ssg-rhel7-ds.xml --dontclean auditd_audispd_remote_daemon_activated`

### CODE REVIEW

**Comment by yuumasato** [comment]
I'm aware of the ansible schema fail and am currently investigating.

File: shared/templates/param_value_in_file/template.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+def preprocess(data, lang):
+    if "sep" not in data:
+        data["sep"] = " = "
+
+    if "prefix_regex" not in data:
+        data["prefix_regex"] = "^\s*"
```
[the remark this answers is not in the document]
Comment (yuumasato) [reply shared/templates/param_value_in_file/template.py]: Fixed.

**Comment by yuumasato** [comment]
> I'm aware of the ansible schema fail and am currently investigating.

This should be solved now.

**Comment by yuumasato** [comment]
The Automatus fails happen because the rules are available for rhel7 only.

**Comment by ggbecker** [comment]
> The Automatus fails happen because the rules are available for rhel7 only.

In case of the [Automatus Fedora](https://github.com/ComplianceAsCode/content/actions/runs/3646753113/jobs/6158230796) check, it uses the RHEL7 content as expected but it fails on the test scenario set up, mainly because the expected file doesn't exist:

From logs:
```
##### auditd_audispd_remote_daemon_activated / wrong_value.fail.sh #####
ssh -o Port=33619 -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null root@localhost cd /root/ssgts/auditd_audispd_remote_daemon_activated; SHARED=/root/ssgts/shared bash -x wrong_value.fail.sh
STDERR: Warning: Permanently added '[localhost]:33619' (ECDSA) to the list of known hosts.
+ sed -i /active/d /etc/audisp/plugins.d/au-remote.conf
sed: can't read /etc/audisp/plugins.d/au-remote.conf: No such file or directory
+ echo 'active = wrong_value'
```

One thing it can be improved is that the test scenario makes sure that the path+file from the template input exists and the test scenario can be properly instantiated

**Review by vojtapolasek — CHANGES_REQUESTED** [review]
Thank you for this new template, it looks very elegant with those macros.
Just small nitpicks, see comments.

File: linux_os/guide/system/auditing/configure_auditd_data_retention/auditd_audispd_remote_daemon_direction/rule.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    <pre>$ sudo service auditd restart</pre>
+
+rationale: |-
+    The auditd service does not include the ability to send audit
+    records to a centralized server for management directly. It does, however,
+    include a plug-in for audit event multiplexor (audispd) to pass audit records
+    to a remote server.
+
+    The direction is dictated by the plugin, and this parameter is used by the multiplexor
+    to understand the directon of events.
```
[the remark this answers is not in the document]
Comment (yuumasato) [reply linux_os/guide/system/auditing/configure_auditd_data_retention/auditd_audispd_remote_daemon_direction/rule.yml]: Thank addressed in 98d0ff5e256cbdfaa3c4afb727962c6dd9dc4372

File: linux_os/guide/system/auditing/configure_auditd_data_retention/auditd_audispd_remote_daemon_activated/rule.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+prodtype: rhel7
+
+title: 'Configure auditd to use audispd''s remote logging daemon'
+
+description: |-
+    To configure the <tt>auditd</tt> service to use the
+    <tt>audisp-remote</tt> plug-in of the <tt>audispd</tt> audit event multiplexor, set
+    the <tt>active</tt> directive in <tt>{{{ audisp_conf_path }}}/plugins.d/au-remote.conf</tt>
+    to <tt>yes</tt>.
+    Restart the <tt>auditd</tt> service:
```
[the remark this answers is not in the document]
Comment (yuumasato) [reply linux_os/guide/system/auditing/configure_auditd_data_retention/auditd_audispd_remote_daemon_activated/rule.yml]: Thank addressed in 98d0ff5e256cbdfaa3c4afb727962c6dd9dc4372

**Comment by vojtapolasek** [comment]
The Openshift CI has some temporary problem. They are working on a fix.

**Comment by yuumasato** [comment]
@vojtapolasek I guess we can merge this. The PR is adding a new template a a new rules applicable for RHEL7 only.
And some of the early commits passed most of the Openshift CI.

**Review by vojtapolasek — APPROVED** [review]

**Comment by vojtapolasek** [comment]
I agree. I hope it would take shorter to fix the issue. I am merging.a

**Pull request merged**